index.bs

```diff
@@ -2921,35 +2920,43 @@ the same operation or constructor.
-            and “required” otherwise.
-        1.  Add to |S| the tuple &lt;|X|, |t|<sub>0..|n|−1</sub>, |o|<sub>0..|n|−1</sub>&gt;.
+        1.  Let |arguments| be the [=list=] of arguments |X| is declared to take.
+        1.  Let |n| be the [=list/size=] of |arguments|.
+        1.  Let |types| be a [=list=] of types.
+        1.  Let |optionality_values| be a list of [=optionality values=].
+        1.  [=list/For each=] |argument| in |arguments|:
+            1.  [=list/Append=] the type of |argument| to |types|.
+            1.  [=list/Append=] "variadic" to |optionality_values| if |argument| is a final, variadic argument,
+                "optional" if |argument| is [=optional argument|optional=],
+                and "required" otherwise.
+        1.  [=set/Append=] the [=tuple=] (|X|, |types|, |optionality_values|) to |S|.
         1.  If |X| is declared to be [=variadic=], then:
-            1.  For every integer |i|, such that |n| ≤ |i| ≤ |m|−1:
-                1.  Let |u|<sub>0..|i|</sub> be a list of types,
-                    where |u|<sub>j</sub> = |t|<sub>j</sub> (for |j| &lt; |n|) and
-                    |u|<sub>j</sub> = |t|<sub>|n|−1</sub> (for |j| ≥ |n|).
-                1.  Let |p|<sub>0..|i|</sub> be a list of [=optionality values=],
-                    where |p|<sub>j</sub> = |o|<sub>j</sub> (for |j| &lt; |n|) and
-                    |p|<sub>j</sub> = “variadic” (for |j| ≥ |n|).
-                1.  Add to |S| the tuple &lt;|X|, |u|<sub>0..|i|</sub>, |p|<sub>0..|i|</sub>&gt;.
-        1.  Initialize |i| to |n|−1.
-        1.  While |i| ≥ 0:
-            1.  If argument |i| of |X| is not
-                [=optional argument|optional=]
-                (i.e., it is not marked as <code>optional</code> and is not
-                a final, variadic argument), then break this loop.
-            1.  Otherwise, add to |S| the tuple &lt;|X|, |t|<sub>0..|i|−1</sub>, |o|<sub>0..|i|−1</sub>&gt;.
+            1.  [=list/For each=] |i| in [=the range=] |n| to |max|−1, inclusive:
+                1.  Let |t| be a [=list=] of types.
+                1.  Let |o| be a [=list=] of [=optionality values=].
+                1.  [=list/For each=] |j| in [=the range=] 0 to |i|, inclusive:
+                    1.  If |j| &lt; |n|, then:
+                        1.  [=list/Append=] |types|[|j|] to |t|.
+                        1.  [=list/Append=] |optionality_values|[|j|] to |o|.
+                    1.  Otherwise:
+                        1.  [=list/Append=] |types|[|n|−1] to |t|.
+                        1.  [=list/Append=] "variadic" to |o|.
+                1.  [=set/Append=] the [=tuple=] (|X|, |t|, |o|) to |S|.
+        1.  Let |i| be |n|−1.
+        1.  [=iteration/While=] |i| ≥ 0:
+            1.  If |arguments|[|i|] is not [=optional argument|optional=]
+                (i.e., it is not marked as "optional" and is not
+                a final, variadic argument), then [=iteration/break=] this loop.
```

Comment:
Just say "break", no need for "this loop" now that Infra gives it a nice definition.